docs/CONTRIBUTING.md

```diff
@@ -35,6 +35,36 @@ We're always looking for contributors.
 
 To run Argo Workflows locally for development: [running locally](running-locally.md).
 
+### Committing
+
+Before you commit code and raise a PR, always run:
+
+```bash
+make pre-commit -B
+```
+
+Please do the following when creating your PR:
+
+* [Sign-off your commits.](https://probot.github.io/apps/dco/)
+* Use [Conventional Commit messages](https://www.conventionalcommits.org/en/v1.0.0/).
+* [Suffix the issue number.](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)
+
+Examples:
+
+```bash
+git commit --signoff -m 'fix: Fixed broken thing. Fixes #1234'
+```
+
+```bash
+git commit --signoff -m 'feat: Added a new feature. Fixes #1234'
+```
+
+TroubleShooting:
+
+* When running `make pre-commit -B`, if you encounter errors like
+  `make: *** [pkg/apiclient/clusterworkflowtemplate/cluster-workflow-template.swagger.json] Error 1`, ensure that you
+  have checked out your code into `$GOPATH/src/github.com/argoproj/argo-workflows`.
```

Comment:
But this pre-commit requires installing the toolchain which are listed in running-locally document.

Reply:
Yea this may make more sense to just link out to `running-locally.md#committing`

Reply:
@terrytangyuan @agilgur5 

Thank you for your review! :)

Based on your feedback, I have kept the Committing-related information in the "running locally" file as is, and added a link to the Committing section within the "running locally" file to the "contributing" file.

This is expected to make it easier for developers to find the commit guidelines for PRs.